powerauth-webflow-resources/src/main/java/io/getlime/security/powerauth/lib/webflow/resource/controller/UserProfileController.java

```diff
@@ -60,13 +63,18 @@ public UserProfileController(DataAdapterClient client, AuthorizationServerTokenS
      */
     @RequestMapping("me")
     public @ResponseBody UserResponse me(OAuth2Authentication authentication) {
+        UserResponse userResponse = new UserResponse();
```

Comment:
I don't like the style of how this code is added very much. It is a minor style issue, but we should create "top objects" first, and proceed to the bottom based on the topology / importance of the properties... I would shoot it like so:

```java
@RequestMapping("me")
public @ResponseBody UserResponse me(OAuth2Authentication authentication) {
    UserResponse userResponse = new UserResponse();

    // Try to fetch user details from the service
    try {
        final ObjectResponse<UserDetailResponse> userDetailResponse = client.fetchUserDetail(authentication.getUserAuthentication().getName());

        // Get additional information stored with the token
        Map<String, Object> additionalInfo = tokenServices.getAccessToken(authentication).getAdditionalInformation();
        
        UserDetailResponse userDetail = userDetailResponse.getResponseObject();
        userResponse.getUser().setId(userDetail.getId());
        userResponse.getUser().setGivenName(userDetail.getGivenName());
        userResponse.getUser().setFamilyName(userDetail.getFamilyName());
        userResponse.getConnection().setLanguage((String) additionalInfo.get(LANGUAGE));
        userResponse.getConnection().setSca((Boolean) additionalInfo.get(SCA));
    } catch (DataAdapterClientErrorException e) {
        // Return dummy user
        userResponse.getUser().setId("anonymousUser");
        userResponse.getUser().setGivenName(null);
        userResponse.getUser().setFamilyName(null);
        userResponse.getConnection().setLanguage("en");
        userResponse.getConnection().setSca(false);
    }
    // Save service information
    userResponse.getService().setApplicationName(webFlowResourcesServerConfiguration.getApplicationName());
    userResponse.getService().setApplicationDisplayName(webFlowResourcesServerConfiguration.getApplicationDisplayName());
    userResponse.getService().setApplicationEnvironment(webFlowResourcesServerConfiguration.getApplicationEnvironment());
    userResponse.getService().setTimestamp(new Date());

    // Return response
    return userResponse;
}
```

Reply:
Sure, let's change it.